File: solutions/c/resistor-color-trio/1/resistor_color_trio.c
#include "resistor_color_trio.h"
#include <stdint.h>
#include <stdlib.h>



uint16_t single_code(resistor_band_t color){
    return (uint16_t) color;    
}

resistor_value_t color_code(resistor_band_t colors[]){
        resistor_value_t res;
        uint16_t digit2 = single_code(colors[1]);
        res.value = (digit2 == 0) ? single_code(colors[0]) : single_code(colors[0])*10 + single_code(colors[1]);
        uint16_t zeros = single_code(colors[2]);
        if (!digit2) 
            zeros++;
        switch(zeros % 3){
            case 1: res.value *= 10;
                    break;
            case 2: res.value *= 100;
                    break;
            default: 
                break;
        }
        res.unit = (ohm_units_t) (zeros/3);
        return res;
}

File: solutions/dart/resistor-color-trio/1/lib/resistor_color_trio.dart
class ResistorColorTrio {
  final List<String> _colors = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];

  String label(List<String> code){
    String unit = "ohms";
    int digit1 = _colors.indexOf(code[0]);
    int digit2 = _colors.indexOf(code[1]);
    int val = (0 == digit2) ? digit1 : 10*digit1 + digit2;
    int zeros = _colors.indexOf(code[2]);
    if (digit2==0)
      zeros++;
    switch(zeros % 3){
        case 1 : val *= 10;
        case 2 : val *= 100;
        default : break;
    }
    switch(zeros ~/ 3){
        case 1: unit = "kilo" + unit;
        case 2: unit = "mega" + unit;
        case 3: unit = "giga" + unit;
      default : break;
    }
    return val.toString() + " " + unit;
  }
}
